src/serializer/ResidualHeapSerializer.js

```diff
@@ -695,6 +696,82 @@ export class ResidualHeapSerializer {
     return t.arrayExpression(initProperties);
   }
 
+  _getPropertyValue(object: any, key: string, serialzeValue: boolean) {
+    if (object != null && object.has(key) === true) {
+      let val = object.get(key);
+
+      if (val !== undefined) {
+        let descriptor = val.descriptor;
+
+        if (descriptor !== undefined) {
+          let descriptorValue = descriptor.value;
+
+          if (descriptorValue !== undefined) {
+            return serialzeValue === true ? this.serializeValue(descriptorValue) : descriptorValue;
+          }
+        }
+      }
+    }
+    return null;
+  }
+
+  _serializeValueReactElement(val: ObjectValue): BabelNodeExpression {
+    let objectProps = val.properties;
+    let typeValue = this._getPropertyValue(objectProps, "type", true);
+    let keyValue = this._getPropertyValue(objectProps, "key", true);
+    let refValue = this._getPropertyValue(objectProps, "ref", true);
+    let propsValue = this._getPropertyValue(objectProps, "props", false);
+
+    invariant(typeValue !== null, "JSXElement type of null");
+
+    let identifier = convertExpressionToJSXIdentifier(typeValue);
+    let attributes = [];
+    let children = [];
+
+    if (keyValue !== null && keyValue.type !== "NullLiteral") {
+      attributes.push(convertKeyValueToJSXAttribute("key", keyValue));
+    }
+
+    if (refValue !== null && refValue.type !== "NullLiteral") {
+      attributes.push(convertKeyValueToJSXAttribute("ref", refValue));
+    }
+
+    if (propsValue instanceof ObjectValue) {
+      this.serializedValues.add(propsValue);
+      // have to case propsValue to ObjectValue or Flow complains that propsValues can be null/undefined
+      for (let [key, propertyBinding] of (propsValue: ObjectValue).properties) {
+        let desc = propertyBinding.descriptor;
+        if (desc === undefined) continue; // deleted
+
+        invariant(key !== "key" && key !== "ref", `"${key}" is a reserved prop name`);
+
+        if (key === "children" && desc.value !== undefined) {
+          let expr = this.serializeValue((desc.value: any));
```

Comment:
If you want to avoid these any casts you can pull `desc.value` out into a local variable before the if statement. Flow doesn't track member accesses as well as locals.